src/main/kotlin/exercise3/task3/SherlockValidatesString.kt

```diff
@@ -29,7 +29,24 @@ package exercise3.task3
  */
 
 internal fun isSherlockValid(s: String): String {
-    TODO("Implement me!!!")
+    val charCounts = s.groupBy { it }.values.map { it.size }
+
+    if (charCounts.toSet().size == 1) {
+        return "YES"
+    }
+
+    if (charCounts.toSet().size == 2) {
+        val minFreq = charCounts.minOrNull() ?: return "NO"
+        val maxFreq = charCounts.maxOrNull() ?: return "NO"
+        val minFreqCounter = charCounts.count { it == minFreq }
```

Comment:
If we know that we have two different numbers and get their min and max values, why is it necessary to use a counter here?
Isn't it enough to check if both values are the same, or that maxFreq - minFreq is equal to 1?